pkgs/development/python-modules/muscima/default.nix

```diff
@@ -33,14 +33,18 @@ buildPythonPackage {
     matplotlib
   ];
 
-  nativeCheckInputs = [
-    pytestCheckHook
-  ];
+  separateChecks = true;
 
-  disabledTestPaths = [
-    # They hard-code the path to the dataset and expect you to edit the test to update it to your value
-    "test/test_dataset.py"
-  ];
+  checkArgs = {
```

Comment:
If we had structuredAttrs, we would not need to nest this in another attr, right?

Reply:
We'd still want a separate argument. We don't want to rebuild the main package when you change some check arguments.

Reply:
I've incorped https://github.com/NixOS/nixpkgs/pull/327264 into this PR and made changes to account for it.

Now the logic is as follows:
- `checkArgs` takes precedence, and allows for passing arbitrary check derivation arguments
- Testing parameters are picked out as appropriate and applied only to the testing derivation in case of separated checks

Reply:
I had to undo the incorp of https://github.com/NixOS/nixpkgs/pull/327264 because of breakages that I'm not able to deal with right now.